Use ruby 3.2.2

crazytonyli opened the pull request:
What says in the title. It should be okay to merge this PR as long as CI is green.

---

- [x] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary.

crazytonyli committed:
Use Ruby 3.2.2

crazytonyli committed:
Update Rubocop

crazytonyli committed:
Use Xcode 14.3.1 on CI

crazytonyli committed:
Update cocoapods to 1.14.3

crazytonyli committed:
Use Xcode 15.1

mokagio commented:
@crazytonyli , interesting failure when validating the `podspec`

```
ERROR \| [tvOS] unknown: Encountered an unknown error (Could not find a `tvos` simulator (valid values: ios, watchos). 
Ensure that Xcode -> Window -> Devices has at least one `tvos` simulator listed or otherwise add one.
```

I suppose the CI image does not download that SDK? 🤔

crazytonyli commented:
@mokagio Yep. We used to install tvOS SDK on the VMs, but removed it because no project runs on tvOS. Well, now we know this project does, sort of...

I guess we have a few options:
1. Remove tvOS support when integrated via CocoaPods ([tvOS is still supported when integrated as Swift package](https://github.com/wordpress-mobile/wpxmlrpc/blob/0.10.0/Package.swift#L7)). If someone who actually uses this library on tvOS apps come yelling at us, we can consider adding it back (or, we can yell back telling them to use SPM).
2. Update the `validate_podspec` utility to accept a `platforms` option, which would allow us to bypass linting on tvOS here.
3. Install tvOS SDK on VMs, which means we'll have to wait for a new VM.

My preferred option is 1, what do you think?

mokagio committed:
Remove tvOS support in CocoaPods

None of our client needs it and supporting it at the CocoaPods level
requires additional infrastructure that we find hard to justify having.

Notice that tvOS is still supported when integrating the library via
Swift Package.

See also
https://github.com/wordpress-mobile/wpxmlrpc/pull/76#issuecomment-1885975334

mokagio committed:
Add entry for tvOS support removal to changelog

mokagio commented:
@crazytonyli 

> My preferred option is 1 [remove tvOS support], what do you think?

I agree. https://github.com/wordpress-mobile/wpxmlrpc/pull/77

crazytonyli committed:
Use Xcode 14.3.1 on CI

crazytonyli committed:
Merge pull request #77 from wordpress-mobile/mokagio/remove-tvos

Remove tvOS support
